### PR TITLE
Interfaces file spaces fix

### LINKED ISSUE #6120: Module interfaces_file ignore option line not starting with a space
### Summary

When the  module interfaces_file look for options in the interfaces file, it use the following  regexp (https://github.com/ansible-collections/community.general/blob/main/plugins/modules/interfaces_file.py#L168):
```python
    patt = re.compile(r'^\s+(?P<option>\S+)\s+(?P<value>\S?.*\S)\s*$')
```

It required each option line to start with at least one space character (`^\s+`). The module ignore option lines not starting with  space. Modifying this option results in duplicate lines.

To fix this issue, replace `+` by `*`:
```python
    patt = re.compile(r'^\s*(?P<option>\S+)\s+(?P<value>\S?.*\S)\s*$')
```

### Issue Type

Bug Report

### Component Name

interfaces_file

### Ansible Version

```console (paste below)
ansible [core 2.14.3]
  config file = /home/olivier/xxxxxxx/ansible.cfg
  configured module search path = ['/home/olivier/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.11/dist-packages/ansible
  ansible collection location = /home/olivier/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.11.1 (main, Dec 31 2022, 10:23:59) [GCC 12.2.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True

```

### Community.general Version

```console (paste below)
# /usr/local/lib/python3.11/dist-packages/ansible_collections
Collection        Version
----------------- -------
community.general 6.4.0  
```

### Configuration

_No response_

### OS / Environment

_No response_

### Steps to Reproduce

The task:
```yaml
- name: Set MTU for eth1
  community.general.interfaces_file:
    iface: "eth1"
    option: mtu
    value: "9000"
    state: present
```

My interfaces file before:
```
iface lo inet loopback
auto lo

auto ens192
iface ens192 inet static                                                                                                                                   
address 10.0.0.1
netmask 255.255.255.0
mtu 1500
```

### Expected Results

```console
iface lo inet loopback
auto lo

auto ens192
iface ens192 inet static                                                                                                                                   
address 10.0.0.1
netmask 255.255.255.0
mtu 9000
```

### Actual Results

```console (paste below)
iface lo inet loopback
auto lo

auto ens192
iface ens192 inet static                                                                                                                                   
address 10.0.0.1
netmask 255.255.255.0
mtu 1500
mtu 9000
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by hryamzik**
##### SUMMARY

Fixes #6120 

##### ISSUE TYPE

- Bugfix Pull Request
- Refactoring Pull Request

##### COMPONENT NAME

interfaces_file

##### ADDITIONAL INFORMATION

Basically the fix was reverting changes from #3328. Added test cases for the issue.

It may also be a good idea to check how tests work for an empty file, not part of this PR of course.

### CODE REVIEW

**Comment by hryamzik** [comment]
- Please add 44 license files!
- You have no changed more than 50 files!

🤣

**Comment by felixfontein** [comment]
bot_skip

**Comment by hryamzik** [comment]
This [issue](https://github.com/ansible-collections/community.general/actions/runs/4321750062/jobs/7543345359) looks valid, I'm trying to debug it locally.

**Comment by hryamzik** [comment]
image
Managed to get unit tests fail with test data from integration tests, that's a little easier to debug.

File: tests/unit/plugins/modules/interfaces_file/test_interfaces_file.py [hunk 1]
```diff
@@ -73,10 +75,24 @@ def compareInterfacesLinesToFile(self, interfaces_lines, path, testname=None):
...
+        testfilepath = os.path.join(golden_output_path, testname)
+        string = json.dumps(ifaces, sort_keys=True, indent=4, separators=(',', ': '))
+        if string and not string.endswith('\n'):
+            string += '\n'
+        goldenstring = string
+        goldenData = ifaces
+        if not os.path.isfile(testfilepath):
+            f = io.open(testfilepath, 'wb')
+            f.write(string.encode())
+            f.close()
```
Comment (felixfontein) [inline tests/unit/plugins/modules/interfaces_file/test_interfaces_file.py:88]:
```suggestion
            with io.open(testfilepath, 'wb') as f:
                f.write(string.encode())
```

File: tests/unit/plugins/modules/interfaces_file/test_interfaces_file.py [hunk 2]
```diff
@@ -73,10 +75,24 @@ def compareInterfacesLinesToFile(self, interfaces_lines, path, testname=None):
...
+        else:
+            with open(testfilepath, 'r') as goldenfile:
+                goldenData = json.load(goldenfile)
+                goldenfile.close()
```
Comment (felixfontein) [inline tests/unit/plugins/modules/interfaces_file/test_interfaces_file.py:92]:
```suggestion
```

**Comment by felixfontein** [comment]
Looks good to me. I'll keep this open for a few more days in case someone else wants to take a look.

**Comment by felixfontein** [comment]
@hryamzik thanks for fixing this!

**Comment by felixfontein** [comment]
I created a manual backport in #6170. There wasn't really a conflict, it was unhappy because the files were added to a directory that was apparently renamed (and it guessed correctly). So it should be fine, but I would appreciate if you @hryamzik could take a quick look at it.
